src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -713,5 +728,45 @@ protected override bool TryParseRequest(ReadResult result, out bool endConnectio
         }
     }
 
+#pragma warning disable CS0618 // Type or member is obsolete
+    private void DetectHttp2Preface(ReadOnlySequence<byte> requestData, BadHttpRequestException ex)
+#pragma warning restore CS0618 // Type or member is obsolete
+    {
+        const int PrefaceLineLength = 18;
```

Comment:
Nit: I think `PrefaceLineLength` should be 16 because `HttpParser.ParseRequestLine()` can throw as soon as "PRI * HTTP/2.0\r\n" is received. In practice, we'll probably always receive the entire `ClientPreface` all at once, but we could just not slice at all if we wanted to rely on that.

```suggestion
        const int PrefaceLineLength = 16;
```